ReactiveUI.Tests/ObservableAsPropertyHelperTest.cs

```diff
@@ -132,4 +138,78 @@ public void ToPropertyShouldFireBothChangingAndChanged()
             Assert.Equal("Baz", resultChanged[1].Value);
         }
     }
+
+    class UnitTestMode : IModeDetector
```

Comment:
pretty minor but this could be a IDisposable helper so one'd write the test like that (so that all test scope is within the test method):

    using (new ProductionMode()) {
        ...
    }

also minor but I believe you can use new PlatformModeDetector() when resetting the mode (it's the default).